File: 8.18.24/src/app/QuestionDto.ts
  export interface AnswerOptionDto {
    optionValue: string;
    nextQuestionId?: number; 
  }

  export interface QuestionDto {
    id: number;                          
    question: string;                    
    serialNumber: number;                
    responseType: string;               
    answerOptions: AnswerOptionDto[];    
    dataType: string;                   
    constraint: string;                  
    constraintValue: string;             
    warningMessage: string;           
    required: boolean;
    nextQuestionId: string;
    userId:any;
  }
  

File: dynamic_backend/DynamicFormServices/Dto/QuestionDto.cs
namespace DynamicFormServices.Dto
{
    public class AnswerOptionDto
    {
        public string OptionValue { get; set; }
        public int? NextQuestionId { get; set; } 
    }

    public class QuestionDto
    {
        public int Id { get; set; }
        public string Question { get; set; }
        public int SerialNumber { get; set; } 
        public string ResponseType { get; set; }
        public List<AnswerOptionDto> AnswerOptions { get; set; }
        public string DataType { get; set; }
        public string Constraint { get; set; } 
        public string ConstraintValue { get; set; } 
        public string WarningMessage { get; set; }
        public bool? Required { get; set; }

        public int UserId { get; set; }

    }
}
